t/rest_server/main_test.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/abcxyz/pkg/logging"
+	"github.com/abcxyz/pkg/renderer"
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestRealMain(t *testing.T) {
+	ctx := logging.WithLogger(context.Background(), logging.TestLogger(t))
+	ctx, done := context.WithCancel(ctx)
+	defer done()
+
+	var realMainErr error
+	finishedCh := make(chan struct{})
+	go func() {
+		defer close(finishedCh)
+		realMainErr = realMain(ctx)
+	}()
+
+	time.Sleep(100 * time.Millisecond) // wait for server startup
+	resp, err := http.Get(fmt.Sprintf("http://localhost:%s/", defaultPort))
```

Comment:
Should `defaultPort` be `*port`? They'll probably be the same, since the user is unlikely to specify `--port=foo` when running test, but technically I think it woulld be more correct to use `*port`.

Reply:
Yes, I agree. Changed to *port